DUI3-DX/Connectors/Rhino/Speckle.Connectors.Rhino7/Operations/Send/RootBaseObjectBuilder.cs

```diff
@@ -0,0 +1,154 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Rhino.DocObjects;
+using Rhino;
+using Speckle.Connectors.DUI.Bindings;
+using Speckle.Core.Models;
+using Speckle.Core.Kits;
+using System.Threading;
+
+namespace Speckle.Connectors.Rhino7.Operations.Send;
+
+/// <summary>
+/// Stateless builder object to turn an <see cref="ISendFilter"/> into a <see cref="Base"/> object
+/// </summary>
+public class RootBaseObjectBuilder
+{
+  private readonly Func<ISpeckleConverter> _converterFactory;
+
+  public RootBaseObjectBuilder(Func<ISpeckleConverter> converterFactory)
+  {
+    _converterFactory = converterFactory;
+  }
+
+  public Base Build(
+    ISendFilter sendFilter,
+    Action<string, double?>? onOperationProgressed = null,
+    CancellationToken ct = default
+  )
+  {
+    List<RhinoObject> rhinoObjects = sendFilter
+      .GetObjectIds()
+      .Select(id => RhinoDoc.ActiveDoc.Objects.FindId(new Guid(id)))
+      .Where(obj => obj != null)
+      .ToList();
+
+    if (rhinoObjects.Count == 0)
+    {
+      throw new InvalidOperationException("No objects were found. Please update your send filter!");
+    }
+
+    ISpeckleConverter converter = _converterFactory();
+
+    converter.SetContextDocument(RhinoDoc.ActiveDoc);
+    Base commitObject = ConvertObjects(rhinoObjects, converter, onOperationProgressed, ct);
+
+    return commitObject;
+  }
+
+  private static Collection ConvertObjects(
+    List<RhinoObject> rhinoObjects,
+    ISpeckleConverter converter,
+    Action<string, double?>? onOperationProgressed = null,
```

Comment:
We should re-evaluate how we report "onOperationProgressed". Is that `double` supposed to be `0 to 1`, `0 to 100`? It was never very clear

Reply:
I just used the existing implementation for this. The progress reporting is actually not working on dui3/alpha at the moment so someone will have to look at this

Reply:
Fixed! 🙌🏼 The issue was using the commands from the BasicBinding in the SendBinding. Since the BasicBinding.Commands was using a different bridge than the send binding... things were not working as intended.